### PR TITLE
Introduce pixi and update Dockerfile

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
Please make PRs from a branch on the repo so we can test them properly.

File: Dockerfile [hunk 1]
```diff
@@ -1,53 +1,30 @@
-FROM mambaorg/micromamba:git-0f27156 AS build-env
-
-ENV PYTHONDONTWRITEBYTECODE=1
+FROM ghcr.io/prefix-dev/pixi:0.26.1 AS build
 USER root
 
 # make sure the install below is not cached by docker
-ADD https://loripsum.net/api /opt/docker/etc/gibberish-to-bust-docker-image-cache
```
Comment (beckermr) [inline Dockerfile:7]: This line prevents docker from caching the steps below during rebuilds. Is there a reason we don't still need it?

Comment (wolfv) [reply Dockerfile:7]: The lockfile should be everything we need to break the cache, no?

File: Dockerfile [hunk 2]
```diff
@@ -1,53 +1,30 @@
...
-COPY / webservices_dispatch_action/
-RUN source /opt/conda/etc/profile.d/conda.sh && \
-    cd webservices_dispatch_action && \
-    conda activate base && \
-    pip install --no-build-isolation -e . && \
-    git clone https://github.com/regro/cf-scripts.git && \
-    cd cf-scripts && \
-    pip install --no-build-isolation -e .
 
 ENTRYPOINT ["/opt/conda/bin/tini", "--", "/opt/docker/bin/entrypoint"]
```
Comment (beckermr) [inline Dockerfile]: You have moved the env, so I am guessing tini somewhere else and this path is wrong?

Comment (wolfv) [reply Dockerfile]: Yes, we would start this using `pixi run` instead. I didn't get there yet :)

**Comment by wolfv** [comment]
I reopened the PR from a branch.

**Pull request closed without merging**